### PR TITLE
Description field should not be mandatory in ANY of the server connec…

### PR DESCRIPTION
**Pull request by dhekra-rouatbi**
**Select the type of change(s) made in this pull request:**
- [X] Bug fix *(Fixes an issue)*
- [ ] New feature *(Adds functionality)*
- [ ] Documentation *(Fix to documentation)*

----------------------------------------------------------------------------------------

Fixes #issue-number

T-iOS-500

### CODE REVIEW

**Comment by ValbuenaG** [comment]
I'm not sure about the task specs. Should it be mandatory for the submission?

File: Tella/Scenes/Nextcloud/ViewModels/NextcloudDraftViewModel.swift [hunk 1]
```diff
@@ -48,6 +36,13 @@ class NextcloudDraftViewModel: DraftMainViewModel {
...
         
         self.objectWillChange.send()
+        
+        DispatchQueue.main.async {
+            self.isValidTitle =  self.title.textValidator()
+            self.isValidDescription = self.description.textValidator()
+            self.objectWillChange.send()
+        }
+
     }
```
Comment (Rimktarii) [inline Tella/Scenes/Nextcloud/ViewModels/NextcloudDraftViewModel.swift:46]: Maybe we can create a function in `DraftMainViewModel` and then call it in the subclasses : 
Here an example : 
create this method inside `DraftMainViewModel` 
   ```
 func validateTitleAndDescription() {
        self.isValidTitle =  self.title.textValidator()
        self.isValidDescription = self.description.textValidator()
        self.objectWillChange.send()
    }

```
and call it here :


` self.validateTitleAndDescription()
`

The same for the other connections

File: Tella/Scenes/Nextcloud/ViewModels/NextcloudDraftViewModel.swift [hunk 2]
```diff
@@ -48,6 +36,13 @@ class NextcloudDraftViewModel: DraftMainViewModel {
         }
         
         self.objectWillChange.send()
+
```
Comment (Rimktarii) [inline Tella/Scenes/Nextcloud/ViewModels/NextcloudDraftViewModel.swift:39]: Maybe we can remove this because it is called  twice

**Review by Rimktarii — CHANGES_REQUESTED** [review]

File: Tella/Scenes/CommonConnectionReport/ViewModels/DraftMainViewModel.swift [hunk 3]
```diff
@@ -136,4 +148,8 @@ class DraftMainViewModel: ObservableObject {
         self.saveReport()
     }
     
+    func validateTitleAndDescription() {
+        self.isValidTitle =  self.title.textValidator()
+        self.isValidDescription = self.description.textValidator()
```
Comment (Rimktarii) [inline Tella/Scenes/CommonConnectionReport/ViewModels/DraftMainViewModel.swift:153]: I think we missed this here :  `self.objectWillChange.send()`

**Review by Rimktarii — CHANGES_REQUESTED** [review]

**Pull request merged**